### ISSUE TITLE
igraph_minimum_size_separators misbehaves with directed graphs

### ISSUE THREAD

**Issue by szhorvat** [igraph/igraph#992]
[The documentation says](http://igraph.org/c/doc/igraph-Separators.html#igraph_minimum_size_separators),

> `graph`: The input graph, it may be directed, **but edge directions will be ignored**.

In practice it seems to return empty results for directed graphs.  I will illustrate with R, but the problem is there in the C function as well.

```
g<-make_graph(c(1,2, 2,3, 3,4, 4,1))
```

```
> min_separators(g)
list()
> min_separators(as.undirected(g))
[[1]]
+ 2/4 vertices:
[1] 2 4

[[2]]
+ 2/4 vertices:
[1] 1 3
```

Can anyone confirm the problem?  Is it a bug or am I misunderstanding something?

**Comment by ntamas** [comment]
I think this is a bug - well, at least in the documentation, because we should have said that the result is undefined for directed graphs. The real problem stems from the fact that while `igraph_minimum_size_separators()` is ignoring edge directions, `igraph_vertex_connectivity()` does not. `igraph_minimum_size_separators()` calls `igraph_vertex_connectivity()` on the input graph to allow for some shortcuts; in particular, if the vertex connectivity of the input graph is 1, `igraph_vertex_connectivity()` falls back to returning the articulation points one by one. That's why you get an empty list for the directed graph: its connectivity is 1 (removing any vertex would make the graph weakly but not strongly connected), but it has no articulation points (removing any vertex would still leave the graph with a single weakly connected component).

I am willing to add an extra check in the C source code that either casts the graph to undirected when the graph is directed (at the expense of making a copy of the graph), or ensures that we always return an empty list. However, releasing a new version of the C core is not up to me, so you would need to depend on the development version with IGraph/M to get this fixed. Alternatively, you can add a check in IGraph/M itself that ensures that only undirected graphs are submitted to this function.

**Comment by szhorvat** [comment]
I have already added a check for this, and I am converting to undirected before passing the graph to this function.  It would be nice to fix it in the core too, though.

I am already basing IGraph/M off (a fork of) the dev version, so that's not a problem at all.  I do not need you to make a release.

**Comment by vtraag** [comment]
I'm trying to think of possible solutions. There are a few possibilities it seems:

1. Simply add a notice to the documentation
2. Convert to an undirected graph (at the costs of making an additional copy).
3. Make sure `igraph_vertex_connectivity` is able to ignore directionality.
4. Make sure that `igraph_minimum_size_separators` is able to handle directionality.

This is also in order of the estimated amount of work that would need to be done. What shall we do for 0.8.0, and what shall we do for 0.9.0? I think it would be best to do option (1) currently for 0.8.0, and implement (4) for 0.9.0. Do you agree?

**Comment by ntamas** [comment]
Yes, (1) is the way to go for 0.8.0 and then make a "proper" fix in 0.9.
